reprounzip-singularity/Singularity_multiple_runs.py

```diff
@@ -0,0 +1,135 @@
+# Singularity unpacker for reprozip 
+import subprocess
+import os
+import sys
+import re
+from shutil import copyfile
+import yaml
+
+#TODO:
+# 1. Logging error for every bash command
+# 2. Using existing reprozip utility functions for common tasks like copying busy box
+# 3. Comments
+# 4. integrating with exisitng unpacking coding design
+# 5. Improving code structure
+
+def extract_reprozip_file(filename):
+    bashCommand = " tar -xf {}".format(filename)
+    process = subprocess.Popen(bashCommand.split(), stdout=subprocess.PIPE)
```

Comment:
you should probably pass a list directly here, rather than `format()` then `split()`